src/async/Task.ts

```diff
@@ -10,7 +10,7 @@ export const Canceled = <State> 4;
  * @param value The value to guard
  */
 export function isTask<T>(value: any): value is Task<T> {
-	return Boolean(isThenable(value) && typeof value.cancel === 'function' && Array.isArray(value.children));
+	return Boolean(isThenable(value) && typeof (<any> value).cancel === 'function' && Array.isArray((<any> value).children));
```

Comment:
It was the `isThenable` narrowing the type, so something like this `return Boolean(value && typeof value.cancel === 'function' && Array.isArray(value.children) && isThenable(value));` works with out the type casting back to `any`